#!/usr/bin/env python
# Raven Visualization offset configuration
PACKAGE = "raven_2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name              Type       Level  Description                             Default  Min   Max
gen.add("shoulder_l",     double_t,  0,     "An offset to add to the shoulder_l",   0,      -50,  50)
gen.add("elbow_l",        double_t,  0,     "An offset to add to the elbow_l",      0,      -50,  50)
gen.add("insertion_l",    double_t,  0,     "An offset to add to the insertion_l",  0,      -50,  50)
gen.add("roll_l",         double_t,  0,     "An offset to add to the roll_l",       0,      -50,  50)
gen.add("wrist_l",        double_t,  0,     "An offset to add to the wrist_l",      0,      -50,  50)
gen.add("grasp1_l",       double_t,  0,     "An offset to add to the grasp1_l",     0,      -50,  50)
gen.add("grasp2_l",       double_t,  0,     "An offset to add to the grasp2_l",     0,      -50,  50)

gen.add("shoulder_r",     double_t,  0,     "An offset to add to the shoulder_r",   0,      -50,  50)
gen.add("elbow_r",        double_t,  0,     "An offset to add to the elbow_r",      0,      -50,  50)
gen.add("insertion_r",    double_t,  0,     "An offset to add to the shoulder_r",   0,      -50,  50)
gen.add("roll_r",         double_t,  0,     "An offset to add to the roll_r",       0,      -50,  50)
gen.add("wrist_r",        double_t,  0,     "An offset to add to the wrist_r",      0,      -50,  50)
gen.add("grasp1_r",       double_t,  0,     "An offset to add to the grasp1_r",     0,      -50,  50)
gen.add("grasp2_r",       double_t,  0,     "An offset to add to the grasp2_r",     0,      -50,  50)

exit(gen.generate(PACKAGE, "raven_2", "Raven2"))
